src/openloco/windows/title_version.cpp

```diff
@@ -0,0 +1,65 @@
+#include "../graphics/colours.h"
+#include "../interop/interop.hpp"
+#include "../openloco.h"
+#include "../ui.h"
+#include "../window.h"
+#include "../windowmgr.h"
+#include <sys/mman.h>
+
+using namespace openloco::interop;
+
+namespace openloco::ui::windows
+{
+    static widget widgets[] = {
+        { 0x1E, 0, 0, 0, 0, 0, 0, 0 }
+    };
+
+    static ui::window_event_list_t events;
+
+    static void draw(window* window);
+
+    uint8_t* buffer;
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+
+    window* open_title_version()
+    {
+        auto window = openloco::ui::windowmgr::create_window(
+            window_type::openloco_version,
+            8,
+            ui::height() - 11 - 8,
+            512,
+            30,
+            (1 << 1) | (1 << 4) | (1 << 5) | (1 << 6),
+            (void*)&events);
+        window->widgets = widgets;
+
+        for (int i = 0; i < 29; i++)
+        {
+            events.pad_00[i * 4 + 0] = 0x35;
+            events.pad_00[i * 4 + 1] = 0xA0;
+            events.pad_00[i * 4 + 2] = 0x42;
+            events.pad_00[i * 4 + 3] = 0x00;
+        }
+
+        events.paint = draw;
+
+        return window;
+    }
+
+    // 0x00439236
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        char str[128] = "OpenLoco 2018.1 (0f2d6ec)";
+
+        {
+            registers regs;
+            regs.cx = window->x;
+            regs.dx = window->y;
```

Comment:
Not sure if you're aware, but resizing the window doesn't seem to affect this.

Reply:
(In this case, `window` refers to the in-game dialog that holds the version info.

Reply:
That makes sense… I guess it's not resizing with the window yet, then? Regardless, you're making nice progress here. 👍